interfaces/cython/cantera/reaction.pyx

```diff
@@ -741,17 +901,114 @@ cdef class ChebyshevReaction(Reaction):
+
+        rxn = TestReaction(equation='H2 + O <=> H + OH',
+                           rate={'A': 38.7, 'b': 2.7, 'Ea': 2.619184e+07},
+                           kinetics=gas)
+
+    Warning: this class is an experimental part of the Cantera API and
+        may be changed or removed without notice.
+    """
+    reaction_type = "elementary-new"
+
+    def __init__(self, equation=None, rate=None, Kinetics kinetics=None,
+                 init=True, **kwargs):
+
+        if init and equation and kinetics:
+
+            if isinstance(rate, dict):
+                coeffs = ['{}: {}'.format(k, v) for k, v in rate.items()]
+            elif isinstance(rate, ArrheniusRate) or rate is None:
+                coeffs = ['{}: 0.'.format(k) for k in ['A', 'b', 'Ea']]
+            else:
+                raise TypeError("Invalid rate definition")
+
+            rate_def = '{{{}}}'.format(', '.join(coeffs))
+            yaml = '{{equation: {}, rate-constant: {}, type: {}}}'.format(
+                equation, rate_def, self.reaction_type)
+            self._reaction = CxxNewReaction(AnyMapFromYamlString(stringify(yaml)),
+                                            deref(kinetics.kinetics))
+            self.reaction = self._reaction.get()
+
+            if isinstance(rate, ArrheniusRate):
+                self.rate = rate
+
+    property rate:
+        """ Get/Set the `Arrhenius` rate coefficient for this reaction. """
+        def __get__(self):
+            cdef CxxTestReaction* r = <CxxTestReaction*>self.reaction
+            return ArrheniusRate.wrap(r.rate())
+        def __set__(self, ArrheniusRate rate):
+            cdef CxxTestReaction* r = <CxxTestReaction*>self.reaction
+            r.setRate(rate._base)
+
+    property allow_negative_pre_exponential_factor:
+        """
+        Get/Set whether the rate coefficient is allowed to have a negative
+        pre-exponential factor.
+        """
+        def __get__(self):
+            cdef CxxElementaryReaction* r = <CxxElementaryReaction*>self.reaction
+            return r.allow_negative_pre_exponential_factor
+        def __set__(self, allow):
+            cdef CxxElementaryReaction* r = <CxxElementaryReaction*>self.reaction
+            r.allow_negative_pre_exponential_factor = allow
+
+
 cdef class InterfaceReaction(ElementaryReaction):
     """ A reaction occurring on an `Interface` (i.e. a surface or an edge) """
-    reaction_type = INTERFACE_RXN
+    reaction_type = "interface"
 
     property coverage_deps:
         """
         Get/Set a dict containing adjustments to the Arrhenius rate expression
         dependent on surface species coverages. The keys of the dict are species
         names, and the values are tuples specifying the three coverage
         parameters ``(a, m, E)`` which are the modifiers for the pre-exponential
-        factor [m, kmol, s units], the temperature exponent [nondimensional],
+        factor [m, kmol, s units], The temperature exponent [nondimensional],
```

Comment:
Why?

Reply:
:man_shrugging: - no idea myself. Not sure how this snuck in!